gateway-ha/src/main/java/com/lyft/data/gateway/ha/router/HaResourceGroupsManager.java

```diff
@@ -160,32 +160,54 @@ public SelectorsDetail createSelector(SelectorsDetail selector) {
    * @return
    */
   @Override
-  public SelectorsDetail updateSelector(SelectorsDetail selector) {
+  public SelectorsDetail updateSelector(SelectorsDetail selector, SelectorsDetail updatedSelector) {
     try {
       connectionManager.open();
-      Selectors model = Selectors.findFirst("resource_group_id = ?", selector.getResourceGroupId());
-
+      Selectors model =
+          Selectors.findFirst(
+              "resource_group_id = ? and priority = ? "
+                  + "and user_regex = ? and source_regex = ? "
+                  + "and query_type = ? and client_tags = ? "
+                  + "and selector_resource_estimate = ?",
+              selector.getResourceGroupId(),
+              selector.getPriority(),
+              selector.getUserRegex(),
+              selector.getSourceRegex(),
+              selector.getQueryType(),
+              selector.getClientTags(),
+              selector.getSelectorResourceEstimate());
       if (model == null) {
-        Selectors.create(new Selectors(), selector);
+        Selectors.create(new Selectors(), updatedSelector);
       } else {
-        Selectors.update(model, selector);
+        Selectors.update(model, updatedSelector);
       }
     } finally {
       connectionManager.close();
     }
-    return selector;
+    return updatedSelector;
   }
 
   /**
    * Search for selector by its resourceGroupId and delete it.
```

Comment:
Is this comment still accurate?

Reply:
Ah, you're right - thanks for catching this!